src/From.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArchTech\Enums;
+
+trait From
+{
+    /**
+     * Gets the Enum by index for "Pure" enums.
+     *
+     * This will not override the `from()` method on BackedEnums
+     *
+     * @throws \ValueError
+     */
+    static public function from(int $case): static
+    {
+        return static::cases()[$case] ?? throw new \ValueError($case . ' is not a valid unit for enum "' . get_called_class() . '"');
+    }
+
+    /**
+     * Gets the Enum by index, if it exists for "Pure" enums.
+     *
+     * This will not override the `tryFrom()` method on BackedEnums
+     */
+    static public function tryFrom(int $case): ?static
+    {
+        return static::cases()[$case] ?? null;
+    }
+
+    /**
+     * Gets the Enum by name.
+     *
+     * @throws \ValueError
+     */
+    static public function fromName(string $case): static
+    {
+        return static::tryFromName($case) ?? throw new \ValueError('"' . $case . '" is not a valid name for enum "' . get_called_class() . '"');
+    }
+
+    /**
+     * Gets the Enum by name, if it exists.
+     */
+    static public function tryFromName(string $case): ?static
+    {
+        $cases = array_filter(
+            static::cases(),
+            fn ($c) => $c->name === $case
+        );
+
+        return array_pop($cases) ?? null;
```

Comment:
If `static::cases()` is not keyed, this can just be `$cases[0]`. If it is keyed, `array_values($cases)[0]`

Reply:
I had tried that, but because `array_filter()` doesn't altter keys, `$cases[0]` might not exist. `array_values($cases)[0]` would (does) also throw a notice about attempting to access an undefined index `0` if the array is empty.

I don't really _like_ using `array_pop()`, but I think that it's probably the most efficient way of getting the "first" item if it exists.

Reply:
> would also throw a notice about attempting to access an undefined index 0 if the array is empty

I think `[0]` is fine when it's followed by `?? null`

Reply:
I must be losing my mind. I could swear that I'd had it that way originally, then changed it because of `Notices` getting thrown. Trying it again now, and no notices.

I'll push up a fix soon.

Reply:
Ahh, the issue was that `array_filter()` was preserving keys, so if the requested `case` wasn't at position `[0]` in the original array, `array_filter()` wouldn't change the index. So:

```php
enum Foo
{
    case foo;
    case bar;
}

$cases = array_filter(Foo::cases(), fn($case) => $case->name === 'bar');

var_dump($cases); // [ 1 => Foo::bar ]
var_dump($cases[0] ?? null); // null
var_dump(array_pop($cases)); // enum(Foo::bar)
```